src/app/room/table/table-card/table-card.component.ts

```diff
@@ -0,0 +1,37 @@
+import {Component, Input} from '@angular/core';
+import {Participant} from "../../../models/participant.model";
+import {map, Observable} from "rxjs";
+import {Card} from "../../../models/card.model";
+import {Store} from "@ngrx/store";
+import {VotingResult} from "../../../models/voting-result.model";
+import * as RoomSelector from "../../../store/room/room.selector";
+
+@Component({
+  selector: 'app-table-card',
+  templateUrl: './table-card.component.html',
+  styleUrls: ['./table-card.component.scss']
+})
+export class TableCardComponent {
+
+  @Input()
+  participant: Participant;
+
+  votingResult$: Observable<VotingResult> = this.store.select(RoomSelector.votingResultSelector);
+  selectedCard$: Observable<Card | undefined> = this.votingResult$.pipe(map(votingResult => {
+    if (votingResult.map instanceof Map) {
+      for (let entry of votingResult.map.entries()) {
+        if (entry[0].nickname == this.participant.nickname) {
```

Comment:
I have to look up a card within a map by the unique field of a participant because if I use the `get()` method, I will get `undefined` even if the map has such participant.

Reply:
Of course, it will return you `undefined`, when you try to get `Card` by `Participant`, because `Participant` object in this component and inside `Map` are different - it is different objects. So, for example, you can use `nickname` as key for `Map` instead of `Patricipant` object.